Mathlib/Topology/Algebra/ProperAction.lean

```diff
@@ -0,0 +1,309 @@
+then there exists `g : G` such that `𝒰.fst` converges to `g`. -/
+theorem properSMul_iff_continuousSMul_ultrafilter_tendsto_t2 [T2Space X] : ProperSMul G X ↔
+    ContinuousSMul G X ∧
+    (∀ 𝒰 : Ultrafilter (G × X), ∀ x₁ x₂ : X,
+    Tendsto (fun gx ↦ ⟨gx.1 • gx.2, gx.2⟩ : G × X → X × X) 𝒰 (𝓝 (x₁, x₂)) →
+    ∃ g : G, Tendsto Prod.fst (𝒰 : Filter (G × X)) (𝓝 g)) := by
+  rw [properSMul_iff_continuousSMul_ultrafilter_tendsto]
+  refine and_congr_right fun hc ↦ ?_
+  congrm ∀ 𝒰 x₁ x₂ hxx, ∃ g, ?_
+  exact and_iff_right_of_imp fun hg ↦ tendsto_nhds_unique
+    (hg.smul ((continuous_snd.tendsto _).comp hxx)) ((continuous_fst.tendsto _).comp hxx)
+
+/-- If `G` acts properly on `X`, then the quotient space is Hausdorff (T2). -/
+@[to_additive "If `G` acts properly on `X`, then the quotient space is Hausdorff (T2)."]
+theorem t2Space_of_ProperSMul (hproper:ProperSMul G X) :
+    T2Space (Quotient (MulAction.orbitRel G X)) := by
+  rw [t2_iff_isClosed_diagonal] -- T2 if the diagonal is closed
+  set R := MulAction.orbitRel G X -- the orbit relation
+  set XmodG := Quotient R -- the quotient
+  set π : X → XmodG := Quotient.mk' -- the projection
+  have hpiopen: IsOpenMap π := -- the projection is open
+    isOpenMap_quotient_mk'_mul
+  have picont : Continuous π := continuous_quotient_mk' -- π is continuous
+  have pisurj : Function.Surjective π := by apply surjective_quotient_mk' -- π is surjective
+  set pipi := Prod.map π π
+  have pipiopen : IsOpenMap pipi := IsOpenMap.prod hpiopen hpiopen -- π × π open
+  have pipisurj : (Function.Surjective (pipi) ) :=  -- π × π surj
+    Function.Surjective.Prod_map pisurj pisurj
+  have pipipquotient := -- π × π is q QuotientMap because open, continuous and surj
+    IsOpenMap.to_quotientMap pipiopen (Continuous.prod_map picont picont) pipisurj
+  rw [<-QuotientMap.isClosed_preimage pipipquotient] -- closed iff preimage closed
+  set gr' := pipi ⁻¹' diagonal (Quotient R) -- preimage of the diag
+  set m : G × X → X × X := fun (g,x) => (g • x, x)
+  set gr := range m -- graph of the orbit relation
+  have r_eq_r' : gr' = gr := by -- the preimage of the diag is the graph of the relation
+    ext ⟨x,y⟩
+    conv_lhs => -- we work only on the left hand side for now
+      rw [mem_preimage]
+      rw [mem_diagonal_iff]
+      change (π x = π y)  --↔ (x, y) ∈ gr
+      rw [Quotient.eq']
+      change ((MulAction.orbitRel G X).Rel x y)-- ↔ (x, y) ∈ gr
+      rw [MulAction.orbitRel_apply]
+      rw [MulAction.orbit]
+      change (∃ g : G, g • y = x)
+    conv_rhs =>
+      rw [mem_range]
+      simp
+    simp [m]
+  rw [r_eq_r']
+  exact hproper.isProperMap_smul_pair'.isClosedMap.closed_range
+
+/-- If a T2 group acts properly on a topological space, then this topological space is T2. -/
+@[to_additive "If a T2 group acts properly on a topological space,
+then this topological space is T2."]
+theorem t2Space_of_properSMul_of_t2Group [h_proper : ProperSMul G X] [T2Space G] : T2Space X := by
```

Comment:
Can this be an `instance`?

Reply:
If I make it an instance I get a "cannot find synthesization order for instance" error. From what I read this is due to the fact that `G` is not mentioned in the instance. Is there a way to get around this?

Reply:
No, indeed. You should leave a lemma then.